e2e/tests-dfx/assetscanister.bash

```diff
@@ -14,6 +14,34 @@ teardown() {
     standard_teardown
 }
 
+@test "authorize and deauthorize work as expected" {
+  PRINCIPAL=$(dfx identity get-principal)
+  install_asset assetscanister
+  dfx_start
+  assert_command dfx deploy
+  [ "$USE_IC_REF" ] && dfx canister update-settings --add-controller "$(dfx canister id e2e_project_frontend)" --all # Dec. 2022: ic-ref has not received the change yet that a canister is always allowed to check it's status
+
+  # deployer is automatically authorized
+  assert_command dfx canister call e2e_project_frontend list_authorized '()'
+  assert_contains "$PRINCIPAL"
+
+  assert_command dfx canister call e2e_project_frontend deauthorize "(principal \"$PRINCIPAL\")"
+  assert_command dfx canister call e2e_project_frontend list_authorized '()'
+  assert_not_contains "$PRINCIPAL"
+
+  # while not authorized, dfx deploy fails
+  echo "new file content" > 'src/e2e_project_frontend/assets/new_file.txt'
+  assert_command_fail dfx deploy
+
+  # canister controllers may always authorize principals, even if they're not authorized themselves
+  assert_command dfx canister call e2e_project_frontend authorize "(principal \"$PRINCIPAL\")"
+  assert_command dfx canister call e2e_project_frontend list_authorized '()'
+  assert_contains "$PRINCIPAL"
```

Comment:
Suggestion: authorize a different identity here, then deploy using it.  The existing tests in this file don't adequately cover authorized/non-authorized functionality.